src/fedora_easyfix/utilities/consumer.py

```diff
@@ -0,0 +1,141 @@
+
+
+class ErraticReturns(object):
+    def parameter_error_return_data(self):
+        return_data = {
+            "status": "FAIL",
+            "salute": "Could not fetch requested information",
+            "reason": "Command string could not be interpreted",
+            "collection_fetched_at": time()
+        }
+        return return_data
+
+    def file_read_error_return_data(self):
+        return_data = {
+            "status": "FAIL",
+            "salute": "Could not fetch requested information",
+            "reason": "Index file could not be accessed",
+            "collection_fetched_at": time()
+        }
+        return return_data
+
+
+class TicketDataRetrieval(object):
+    def __init__(self):
+        self.filename = "tickdata.json"
+        with open(self.filename, "r") as self.fileobjc:
+            self.dictcont = loads(self.fileobjc.read())
+
+    def preliminary_information(self):
+        try:
+            return_data = {
+                "status": "PASS",
+                "forges": {},
+                "collection_fetched_at": 0.0
+            }
+            for forge in self.dictcont["forges"].keys():
+                forge_info = {
+                    "name": forge,
+                    "repository_list": {}
+                }
+                for repository in self.dictcont["forges"][forge].keys():
+                    repository_info = {
+                        "name": repository,
+                        "description": self.dictcont["forges"][forge][repository]["description"],
+                        "contact": self.dictcont["forges"][forge][repository]["contact"],
+                        "issue_list": {}
+                    }
+                    for issue in self.dictcont["forges"][forge][repository]["ticket_list"].keys():
+                        issue_info = {
+                            "title": self.dictcont["forges"][forge][repository]["ticket_list"][issue]["title"],
+                            "labels": self.dictcont["forges"][forge][repository]["ticket_list"][issue]["labels"]
+                        }
+                        repository_info["issue_list"][issue] = issue_info
+                    forge_info[repository] = repository_info
+                return_data["forges"][forge] = forge_info
+            return_data["collection_fetched_at"] = time()
+        except Exception as expt:
+            return_data = {
+                "status": "FAIL",
+                "salute": "Could not fetch preliminary information",
+                "reason": str(expt),
+                "collection_fetched_at": time()
+            }
+        return return_data
+
+    def repository_information(self, forge, repository):
+        try:
+            return_data = {
+                "status": "PASS",
+                "information": {
+                    "name": repository,
+                    "forge": forge,
+                    "ticket_count": self.dictcont["forges"][forge][repository]["ticket_count"],
```

Comment:
You access `self.dictcont["forges"][forge][repository]` pretty often, I'd use a variable to shortcut this, à la:

````
def repository_information(self, forgename, reponame):
    try:
        repo = self.dictcont["forges"][forge][reponame]
        return_data = {
            "name": reponame,
            "forge": forgename,
            "ticket_count": repo["ticket_count"],
            "contact": repo["contact"],
            ...
````

Reply:
Ah, yes. Makes sense.